File: java/freeman/command/macro/TV.java
public class TV {
    boolean on;
    int volume;
    int channel;
    public TV(){
        volume=0;
        on=false;
        channel=0;
    }
    public void setVolume(int volume){
        this.volume=volume;
    }
    public void setChannel(int channel){
        this.channel=channel;
    }
    public void on() {
        if (on){
            System.out.println("TV is already works");
        }
        else{
            on=true;
            System.out.println("On tv!");
        }
	
    }
    public void off() {
	if (on){
            on=false;
            System.out.println("Switch off TV");
        }
        else{
            System.out.println("TV is not working");
        }
    }
    public String toString(){
        String status;
        if (on){
           status="TV status:on. Volume="+volume+" current channel="+channel;
        }
        else{
           status="TV status:off";
        }
        return status;
    }
}

File: php/oop-study/lesson4/Facade/devices/TV.php
<?php
class TV{
	private $on;
    private $volume;
    private $channel;
    public function __construct(){
        $this->volume=0;
        $this->on=false;
        $this->channel=0;
    }
    public function setVolume($volume){
        $this->volume=$volume;
    }
    public function setChannel($channel){
        $this->channel=$channel;
    }
    public function on() {
        if ($this->on){
            echo "TV is already works<br>";
        }
        else{
            $this->on=true;
            echo "On tv!<br>";
        }
	
    }
    public function off() {
		if ($this->on){
            $this->on=false;
            echo "Switch off TV<br>";
        }
        else{
            echo "TV is not working";
        }
    }   
} 
?>